core/schema/git.graphqls

```diff
@@ -55,4 +55,7 @@ type GitRepository {
 type GitRef {
   "The filesystem tree at this ref."
   tree(sshKnownHosts: String, sshAuthSocket: SocketID): Directory!
+
+  "The resolved commit id at this ref."
+  commit(sshKnownHosts: String, sshAuthSocket: SocketID): String!
```

Comment:
:eyes: is there a reason that we set ssh params here, and not at the top-level?

It feels like it would be more reasonable to lift these to the `git` resolver on `Query`, since these logically apply to the entire git repository, and not just individual operations on a `GitRef`.

That would likely be a breaking change though, so I haven't done that right away.

Reply:
I agree moving those settings up makes a ton of sense, but how much trouble (in terms of maintainability) would it be to support it in both parts of the API? If it's not bad, I'd prefer to just do that. While we should still be willing to make breaking changes (pre v1.0 anyways), I think we should avoid it if reasonable+simple alternatives exist.

Reply:
Sounds good to me - we can deprecate the lower-level ones, but keep it on `tree` for now.